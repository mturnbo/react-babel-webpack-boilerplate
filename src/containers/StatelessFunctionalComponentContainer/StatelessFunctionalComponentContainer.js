import React from 'react';
import StatelessFunctionalComponent from 'components/StatelessFunctionalComponent';
import data from '../../data/data.json';
import './StatelessFunctionalComponentContainer.scss';

class StatelessFunctionalComponentContainer extends React.Component {
  render() {
    const componentList = data.map((entry, index) =>
      <StatelessFunctionalComponent title={entry.title} content={entry.content} key={index.toString()} />
    );

    return (
      <div className="stateless-functional-component-container">
        <h2>Stateless Functional Component Container</h2>
        {componentList}
      </div>
    );
  }
}

export default StatelessFunctionalComponentContainer;
