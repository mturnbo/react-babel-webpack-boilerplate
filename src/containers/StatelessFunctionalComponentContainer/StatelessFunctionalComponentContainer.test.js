import React from 'react';
import { mount } from 'enzyme';
import StatelessFunctionalComponentContainer from './index';

const wrapper = mount(<StatelessFunctionalComponentContainer />);

test('StatelessFunctionalComponent Container Renders', () => {
  expect(wrapper).toMatchSnapshot();
});
