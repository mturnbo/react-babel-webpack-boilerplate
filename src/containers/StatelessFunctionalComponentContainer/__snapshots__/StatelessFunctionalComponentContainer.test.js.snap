// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatelessFunctionalComponent Container Renders 1`] = `
<StatelessFunctionalComponentContainer>
  <div
    className="stateless-functional-component-container"
  >
    <h2>
      Stateless Functional Component Container
    </h2>
    <StatelessFunctionalComponent
      content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
      key="0"
      title="Widget"
    >
      <div
        className="stateless-functional-component"
      >
        <div
          className="header"
        >
          Widget
        </div>
        <div
          className="content"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
        <div
          className="button-container"
        >
          <Button
            label="Widget Button"
            onClick={[Function]}
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Widget Button
            </button>
          </Button>
        </div>
      </div>
    </StatelessFunctionalComponent>
    <StatelessFunctionalComponent
      content="Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt."
      key="1"
      title="Thingee"
    >
      <div
        className="stateless-functional-component"
      >
        <div
          className="header"
        >
          Thingee
        </div>
        <div
          className="content"
        >
          Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.
        </div>
        <div
          className="button-container"
        >
          <Button
            label="Thingee Button"
            onClick={[Function]}
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Thingee Button
            </button>
          </Button>
        </div>
      </div>
    </StatelessFunctionalComponent>
    <StatelessFunctionalComponent
      content="Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae."
      key="2"
      title="Whatchamacallit"
    >
      <div
        className="stateless-functional-component"
      >
        <div
          className="header"
        >
          Whatchamacallit
        </div>
        <div
          className="content"
        >
          Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae.
        </div>
        <div
          className="button-container"
        >
          <Button
            label="Whatchamacallit Button"
            onClick={[Function]}
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Whatchamacallit Button
            </button>
          </Button>
        </div>
      </div>
    </StatelessFunctionalComponent>
    <StatelessFunctionalComponent
      content="At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga."
      key="3"
      title="Doohickey"
    >
      <div
        className="stateless-functional-component"
      >
        <div
          className="header"
        >
          Doohickey
        </div>
        <div
          className="content"
        >
          At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.
        </div>
        <div
          className="button-container"
        >
          <Button
            label="Doohickey Button"
            onClick={[Function]}
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Doohickey Button
            </button>
          </Button>
        </div>
      </div>
    </StatelessFunctionalComponent>
    <StatelessFunctionalComponent
      content="Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur."
      key="4"
      title="Thingamajig"
    >
      <div
        className="stateless-functional-component"
      >
        <div
          className="header"
        >
          Thingamajig
        </div>
        <div
          className="content"
        >
          Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur.
        </div>
        <div
          className="button-container"
        >
          <Button
            label="Thingamajig Button"
            onClick={[Function]}
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Thingamajig Button
            </button>
          </Button>
        </div>
      </div>
    </StatelessFunctionalComponent>
    <StatelessFunctionalComponent
      content="Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur?"
      key="5"
      title="Doodad"
    >
      <div
        className="stateless-functional-component"
      >
        <div
          className="header"
        >
          Doodad
        </div>
        <div
          className="content"
        >
          Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur?
        </div>
        <div
          className="button-container"
        >
          <Button
            label="Doodad Button"
            onClick={[Function]}
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Doodad Button
            </button>
          </Button>
        </div>
      </div>
    </StatelessFunctionalComponent>
    <StatelessFunctionalComponent
      content="Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt."
      key="6"
      title="Thingamabob"
    >
      <div
        className="stateless-functional-component"
      >
        <div
          className="header"
        >
          Thingamabob
        </div>
        <div
          className="content"
        >
          Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.
        </div>
        <div
          className="button-container"
        >
          <Button
            label="Thingamabob Button"
            onClick={[Function]}
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Thingamabob Button
            </button>
          </Button>
        </div>
      </div>
    </StatelessFunctionalComponent>
  </div>
</StatelessFunctionalComponentContainer>
`;
